internal/redact/redact.go

```diff
@@ -33,3 +35,57 @@ func FromContext(ctx context.Context) (bool, string) {
 	reason, ok := ctx.Value(redactKey).(string)
 	return ok, reason
 }
+
+// Error redacts potentially sensitive query parameter values in the URL from the error's message.
+//
+// If the error is a *url.Error, this returns a *url.Error with the URL redacted.
+// Any other error type, or nil, is returned unchanged.
+func Error(err error) error {
+	if err == nil {
```

Comment:
`errors.As` does a nil check if you want to be lazier.

Reply:
I always want to be lazier 🏖️